src/ROOTFrameReader.cc

```diff
@@ -37,6 +37,22 @@ std::unique_ptr<ROOTFrameData> ROOTFrameReader::readNextEntry(const std::string&
   if (!catInfo.chain) {
     return nullptr;
   }
+
+  return readEntry(catInfo);
+}
+
+std::unique_ptr<ROOTFrameData> ROOTFrameReader::readEntry(const std::string& name,
+                                                          const unsigned entNum) {
+  auto& catInfo = getCategoryInfo(name);
+  if (!catInfo.chain) {
+    return nullptr;
+  }
```

Comment:
I think this check could also be moved to the private `readEntry` as that would avoid having it duplicated in two places.

Reply:
Moved to private method